From ee17ebe4968012781ba3c80eaf938c2eb1703e72 Mon Sep 17 00:00:00 2001
From: Hunter Sadler <auro@hsadler.com>
Date: Sat, 21 Mar 2026 19:47:29 -0600
Subject: [PATCH 5/7] Fix large agent step-frame requests

---
 command.c | 2 +-
 runloop.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/command.c b/command.c
index 89f4baef05..239ec919e0 100644
--- a/command.c
+++ b/command.c
@@ -794,7 +794,7 @@ bool command_step_frame(command_t *cmd, const char *arg)
       return false;
 
    runloop_st->flags &= ~RUNLOOP_FLAG_PAUSED;
-   runloop_st->run_frames_and_pause = (frames > INT8_MAX) ? INT8_MAX : (int8_t)frames;
+   runloop_st->run_frames_and_pause = (frames > INT32_MAX) ? INT32_MAX : (int32_t)frames;
 
    _len  = strlcpy(reply, "STEP_FRAME ", sizeof(reply));
    _len += snprintf(reply + _len, sizeof(reply) - _len, "%u\n", frames);
diff --git a/runloop.h b/runloop.h
index ae040c2d7b..43d1a8cf52 100644
--- a/runloop.h
+++ b/runloop.h
@@ -269,7 +269,7 @@ struct runloop
 
    uint32_t flags;
    uint8_t pending_disk_control_insert;
-   int8_t run_frames_and_pause;
+   int32_t run_frames_and_pause;
 
    char runtime_content_path_basename[PATH_MAX_LENGTH];
 #ifdef HAVE_SCREENSHOTS
-- 
2.54.0

