From 00dc81041d1082114cb3e1ebfb4738a7c0edb131 Mon Sep 17 00:00:00 2001
From: Hunter Sadler <auro@hsadler.com>
Date: Sun, 22 Mar 2026 00:35:07 -0600
Subject: [PATCH 7/7] Add RetroArch stdin ping command

---
 command.c | 16 ++++++++++++++++
 command.h |  2 ++
 2 files changed, 18 insertions(+)

diff --git a/command.c b/command.c
index 4cfcdae3f4..e981c80257 100644
--- a/command.c
+++ b/command.c
@@ -780,6 +780,22 @@ bool command_set_pause(command_t *cmd, const char *arg)
    return true;
 }
 
+bool command_ping(command_t *cmd, const char *arg)
+{
+   char reply[32];
+   size_t _len = 0;
+
+   (void)arg;
+
+   if (!cmd || !cmd->replier)
+      return false;
+
+   _len  = strlcpy(reply, "PING ", sizeof(reply));
+   _len += strlcpy(reply + _len, "OK\n", sizeof(reply) - _len);
+   cmd->replier(cmd, reply, _len);
+   return true;
+}
+
 bool command_step_frame(command_t *cmd, const char *arg)
 {
    char reply[64];
diff --git a/command.h b/command.h
index 33ad2884cd..c5539c2b74 100644
--- a/command.h
+++ b/command.h
@@ -430,6 +430,7 @@ bool command_version(command_t *cmd, const char* arg);
 bool command_get_status(command_t *cmd, const char* arg);
 bool command_get_config_param(command_t *cmd, const char* arg);
 bool command_show_osd_msg(command_t *cmd, const char* arg);
+bool command_ping(command_t *cmd, const char *arg);
 bool command_set_pause(command_t *cmd, const char *arg);
 bool command_step_frame(command_t *cmd, const char *arg);
 bool command_set_input_port(command_t *cmd, const char *arg);
@@ -455,6 +456,7 @@ static const struct cmd_action_map action_map[] = {
    { "SET_SHADER",       command_set_shader,       "<shader path>" },
 #endif
    { "VERSION",          command_version,          "No argument"},
+   { "PING",             command_ping,             "No argument" },
    { "GET_STATUS",       command_get_status,       "No argument" },
    { "GET_CONFIG_PARAM", command_get_config_param, "<param name>" },
    { "SHOW_MSG",         command_show_osd_msg,     "No argument" },
-- 
2.54.0

